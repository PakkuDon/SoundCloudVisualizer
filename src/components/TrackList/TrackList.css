.root {
  overflow-y: auto;
  /* FIXME Hack to prevent tracklist from overflowing past container height */
  max-height: 90vh;
}
