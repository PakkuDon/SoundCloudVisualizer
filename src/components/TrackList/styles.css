.root {
  overflow-y: auto;
  flex: 1 1 auto;
}

.title {
  font-size: 1.2em;
}
