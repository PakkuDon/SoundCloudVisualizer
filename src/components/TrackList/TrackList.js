import React from "react"
import PropTypes from "prop-types"
import { SoundCloudTrack } from "../../propTypes"
import TrackListItem from "../TrackListItem"
import styles from "./TrackList.css"

const TrackList = ({ onTrackDelete, onTrackQueue, onTrackSelect, tracks }) => (
  <div className={styles.root}>
    {tracks
      .slice()
      .reverse()
      .map((track) => (
        <TrackListItem
          key={`track-${track.id}`}
          onDelete={onTrackDelete}
          onQueue={onTrackQueue}
          onSelect={onTrackSelect}
          track={track}
        />
      ))}
  </div>
)

TrackList.propTypes = {
  onTrackDelete: PropTypes.func,
  onTrackQueue: PropTypes.func,
  onTrackSelect: PropTypes.func,
  tracks: PropTypes.arrayOf(SoundCloudTrack),
}

TrackList.defaultProps = {
  tracks: [],
}

export default TrackList
