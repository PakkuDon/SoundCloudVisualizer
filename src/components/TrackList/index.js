import React from 'react'
import PropTypes from 'prop-types'
import { SoundCloudTrack } from '../../propTypes'
import TrackListItem from '../TrackListItem'
import styles from './styles.css'

const TrackList = ({ onTrackDelete, onTrackSelect, tracks }) => (
  <React.Fragment>
    <div className={styles.root}>
      {tracks.slice().reverse().map(track => (
        <TrackListItem
          key={`track-${track.id}`}
          onDelete={onTrackDelete}
          onSelect={onTrackSelect}
          track={track}
        />
      ))}
    </div>
  </React.Fragment>
)

TrackList.propTypes = {
  onTrackDelete: PropTypes.func,
  onTrackSelect: PropTypes.func,
  tracks: PropTypes.arrayOf(SoundCloudTrack),
}

TrackList.defaultProps = {
  tracks: [],
}

export default TrackList
