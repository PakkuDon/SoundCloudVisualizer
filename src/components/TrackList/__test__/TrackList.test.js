import React from "react"
import { render } from "@testing-library/react"

import TrackList from "../TrackList"
import TrackFactory from "../../../testHelpers/trackFactory"

describe("TrackList", () => {
  it("renders list of tracks", () => {
    const tracks = [
      TrackFactory.create({ id: 1 }),
      TrackFactory.create({ id: 2 }),
    ]
    const { container } = render(<TrackList tracks={tracks} />)

    expect(container).toMatchSnapshot()
  })
})
