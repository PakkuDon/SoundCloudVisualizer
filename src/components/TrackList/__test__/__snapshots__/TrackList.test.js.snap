// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList renders list of tracks 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <div>
        <a
          href="https://soundcloud.com/user2835985"
          rel="noopener noreferrer"
          target="_blank"
        >
          Bob
        </a>
      </div>
      <div>
        Hello World
      </div>
      <div />
      <div>
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          Permalink
        </a>
      </div>
      <div
        class="actions"
      />
    </div>
    <div
      class="root"
    >
      <div>
        <a
          href="https://soundcloud.com/user2835985"
          rel="noopener noreferrer"
          target="_blank"
        >
          Bob
        </a>
      </div>
      <div>
        Hello World
      </div>
      <div />
      <div>
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          Permalink
        </a>
      </div>
      <div
        class="actions"
      />
    </div>
  </div>
</div>
`;
